src/components/DateInput.jsx

```diff
@@ -169,21 +171,15 @@ class DateInput extends React.Component {
       readOnly,
       openDirection,
       verticalSpacing,
+      inputHeight,
       styles,
-      theme: { reactDates },
     } = this.props;
 
     const value = displayValue || dateString || '';
     const screenReaderMessageId = `DateInput__screen-reader-message-${id}`;
 
     const withFang = showCaret && focused;
 
-    const {
-      font: { input: { lineHeight } },
-      spacing: { inputPadding, displayTextPaddingVertical },
-    } = reactDates;
-    const inputHeight = getInputHeight({ lineHeight, inputPadding, displayTextPaddingVertical });
```

Comment:
i'm confused; how can all this logic be replaced by just passing down a prop? I'm fine with it being overrideable, but wouldn't the default need this same logic?

Reply:
Not really because once you set the height to a fixed value, this padding doesn't affect it anymore. Although I guess that the `displayTextPadding` might still increase the whole thing... it's not really being used anymore, so this might be an opportunity just to axe it (it's kind of a weird pattern that doesn't make terribly much sense now that we have a real input).

Reply:
Nope, I'm definitely wrong. Let me reinvestigate!